src/combinatorics/simple_graph/basic.lean

```diff
@@ -355,16 +355,107 @@ lemma complete_graph_is_regular [decidable_eq V] :
 by { intro v, simp }
 
 /--
-The minimum degree of all vertices
+The minimum degree of all vertices (and `0` if there are no vertices).
 -/
-def min_degree (G : simple_graph V) [nonempty V] [decidable_rel G.adj] : ℕ :=
-finset.min' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty _)
+def min_degree [decidable_rel G.adj] : ℕ :=
+option.get_or_else (univ.image (λ v, G.degree v)).min 0
 
 /--
-The maximum degree of all vertices
+There exists a vertex of minimal degree. Note the assumption of being nonempty is necessary, as
+the lemma implies there exists a vertex.
 -/
-def max_degree (G : simple_graph V) [nonempty V] [decidable_rel G.adj] : ℕ :=
-finset.max' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty _)
+lemma exists_minimal_degree_vertex [decidable_rel G.adj] [nonempty V] :
+  ∃ v, G.degree v = G.min_degree :=
```

Comment:
I noticed that when using `cases` on this and `exists_maximal_degree_vertex`, the rewrite always goes backwards. Would it be more ergonomic to reverse the equalities in their statements? I guess the idea would be that you'd probably only ever be using this lemma to introduce a vertex whose degree is `G.min_degree`, so you'd most likely be eliminating `G.min_degree` in favor of `G.degree v`.

Reply:
I think there are use cases for this lemma outside of that, but I agree it's probably a good idea to reverse the equalities, I'll amend. Thanks!

Reply:
Changed. Thanks for the suggestion!